modular_skyrat/modules/ash_chemical_centrifuge/code/ash_centrifuge.dm

```diff
@@ -0,0 +1,22 @@
+/obj/item/reagent_containers/glass/primitive_centrifuge
+	name = "primitive centrifuge"
+	desc = "A small cup that allows a person to slowly spin out liquids they do not desire."
+	icon = 'modular_skyrat/modules/ash_chemical_centrifuge/icons/chemical.dmi'
+	icon_state = "primitive_centrifuge"
+
+/obj/item/reagent_containers/glass/primitive_centrifuge/examine()
+	. = ..()
+	. += span_notice("Ctrl + Click to select chemicals to remove.")
+
+/obj/item/reagent_containers/glass/primitive_centrifuge/CtrlClick(mob/user)
+	var/datum/user_input = tgui_input_list(user, "Select which chemical to remove.", "Removal Selection", reagents.reagent_list)
+	if(!user_input)
+		to_chat(user, span_warning("A selection was not made."))
+		return
+	if(!do_after(user, 2 SECONDS, target = src))
```

Comment:
I'd put at least 5 seconds, this is ghetto.

That being said, I'd also check to make sure that this doesn't allow you to actually change the purity of chems, because that'd probably break a lot of balance in that regard.